src/mpi/comm/commutil.c

```diff
@@ -35,14 +35,122 @@ MPIR_Object_alloc_t MPIR_Comm_mem = {
 
 /* Communicator creation functions */
 struct MPIR_Commops *MPIR_Comm_fns = NULL;
-struct MPIR_Comm_hint_fn_elt {
-    char name[MPI_MAX_INFO_KEY];
+static int MPIR_Comm_commit_internal(MPIR_Comm * comm);
+
+/* Communicator hint functions */
+/* For balance of simplicity and feature, we'll internally use integers for both keys
+ * and values, and provide facilities to translate from and to string-based infos.
```

Comment:
Will this restrict us from using any other types, e.g., floating point hint values? If so, do we want to lose that flexibility?

Reply:
It can be extended of course.

To add a floating point type, it is necessary to make the `hints` array of unions, and every reference need reference the member of the union. Not a big deal, but lets delay that until it is necessary.